project-confero/confero
# Chores/setup db

# Problem

We need a database

# Solution

Set up a database, some models to exercise it, and some docker config to help use it.

*Note*: This probably won't deploy until we get RDS set up, including putting the right env vars in EC2

## Change summary

- For postgres support, add psycopg to django, and build it with docker
- Set up docker-compose for easier dev setup
- Add a couple models and pages to the Django app to show the DB is working
- Documentation
- Add a PR template to remind me to document things

## Steps to Verify

1. Follow the README to start up the database with docker-compose and the app locally
1. Setup the `.env` file
1. (also Follow the README to the while app with docker-compose)
1. Go to localhost:8000 to see the app running
1. Go to http://127.0.0.1:8000/admin/
1. Follow the [Django Admin Docs](https://docs.djangoproject.com/en/2.1/intro/tutorial02/#introducing-the-django-admin) to set up admin users
1. Create a superuser and normal user
1. Login as normal user
1. Create a Politician and some related Campaigns.
1. Go back to http://127.0.0.1:8000
1. See your Politicians
1. Click on a Politician
1. See their campaigns

## Screenshot (Optional)

![screen shot 2019-02-12 at 12 59 29 pm](https://user-images.githubusercontent.com/5178425/52667589-0f809c80-2ec6-11e9-8eae-7d94a90036e8.png)

# For Discussion

I forgot what a pain local docker development was (because we stopped doing it on my current project)
This article sums up my experience with it pretty well: https://medium.com/canal-tech/why-using-docker-in-your-local-dev-environment-is-probably-not-a-great-idea-3836c6823d60
